main_test.go

```diff
@@ -30,19 +30,20 @@ func (c *ComponentTest) InitializeScenario(godogCtx *godog.ScenarioContext) {
 
 	apiFeature := apiComponent.InitAPIFeature()
 
-	godogCtx.BeforeScenario(func(*godog.Scenario) {
+	godogCtx.Before(func(ctx context.Context, sc *godog.Scenario) (context.Context, error) {
 		apiFeature.Reset()
 		c.AuthFeature.Reset()
 		apiComponent.Reset()
+		return context.Background(), nil
```

Comment:
Should the return use `ctx` not `context.Background()`?

```suggestion
		return ctx, nil
```

Reply:
Also on line 45

Reply:
The ctx is context.background() itself which is different from godog context. But yes since you are passing the ctx, just using that here would make sense. So fixed in update 1.